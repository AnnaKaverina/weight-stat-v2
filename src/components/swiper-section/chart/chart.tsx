import React from 'react';
import { Bar } from '../bar';
import { IBarItem } from '../../../interfaces/user-data.interface';
import { Wrapper } from './chart.style';

export function Chart({ values }: { values: Array<IBarItem> }) {
	const renderBars = (values: Array<IBarItem>) => {
		return values.map((barItem) => {
			return <Bar barItem={barItem} key={barItem.title} />;
		});
	};

	return (
		<Wrapper>
			{renderBars(values)}
		</Wrapper>
	);
}
