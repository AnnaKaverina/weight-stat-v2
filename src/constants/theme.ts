export default {
	colors: {
		background: '#fcf3ef',
		font: '#725f59',
		bar: {
			background: '#ffe6d6',
			levelBackground: '#c77d72',
			overflowBackground: '#cb2910',
		},
	},
};
