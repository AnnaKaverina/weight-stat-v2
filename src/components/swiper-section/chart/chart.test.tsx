import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Chart } from './chart';
import theme from '../../../constants/theme';
import { ThemeProvider } from 'styled-components';

describe('chart render', () => {
	it('Chart component should be in the document', () => {
		const values = [
			{
				title: 'Calories',
				value: 2000,
				maxValue: 1500
			},
			{
				title: 'Fats',
				value: 800,
				maxValue: 800
			},
			{
				title: 'Proteins',
				value: 700,
				maxValue: 900
			},
			{
				title: 'Carbohydrates',
				value: 400,
				maxValue: 700
			}
		];
		const { getByText, container } = render(
			<ThemeProvider theme={theme}>
				<Chart values={values} />
			</ThemeProvider>
		);
		const element = getByText('Calories');
		expect(element).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
