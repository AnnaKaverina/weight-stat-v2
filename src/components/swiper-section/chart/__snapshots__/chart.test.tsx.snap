// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chart render Chart component should be in the document 1`] = `
<div>
  <div
    class="sc-iBYQkv bqmDcZ"
  >
    <div
      class="sc-bcXHqe cNKIOy"
    >
      <span
        class="sc-dkrFOg gUajxP"
      >
        Calories
      </span>
      <div
        class="sc-gswNZR bRFjDI"
      >
        <div
          class="sc-hLBbgP kdVVmG"
        />
        <div
          class="sc-gKPRtg jtBuTa"
        />
        <span
          class="sc-eDvSVe eSXpoT"
        >
          2000
        </span>
        <span
          class="sc-jSUZER jKGhqI"
        >
          1500
        </span>
      </div>
    </div>
    <div
      class="sc-bcXHqe cNKIOy"
    >
      <span
        class="sc-dkrFOg gUajxP"
      >
        Fats
      </span>
      <div
        class="sc-gswNZR bRFjDI"
      >
        <div
          class="sc-hLBbgP kdVVmG"
        />
        <div
          class="sc-gKPRtg cQBeSx"
        />
        <span
          class="sc-eDvSVe eSXpoT"
        >
          800
        </span>
        <span
          class="sc-jSUZER jKGhqI"
        >
          800
        </span>
      </div>
    </div>
    <div
      class="sc-bcXHqe cNKIOy"
    >
      <span
        class="sc-dkrFOg gUajxP"
      >
        Proteins
      </span>
      <div
        class="sc-gswNZR bRFjDI"
      >
        <div
          class="sc-hLBbgP wJgpY"
        />
        <div
          class="sc-gKPRtg cQBeSx"
        />
        <span
          class="sc-eDvSVe RCgGL"
        >
          700
        </span>
        <span
          class="sc-jSUZER jKGhqI"
        >
          900
        </span>
      </div>
    </div>
    <div
      class="sc-bcXHqe cNKIOy"
    >
      <span
        class="sc-dkrFOg gUajxP"
      >
        Carbohydrates
      </span>
      <div
        class="sc-gswNZR bRFjDI"
      >
        <div
          class="sc-hLBbgP getOxh"
        />
        <div
          class="sc-gKPRtg cQBeSx"
        />
        <span
          class="sc-eDvSVe iCdyro"
        >
          400
        </span>
        <span
          class="sc-jSUZER jKGhqI"
        >
          700
        </span>
      </div>
    </div>
  </div>
</div>
`;
